modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj

```diff
@@ -466,9 +466,9 @@
   table e.g.
 
     `table`.`field` -> `dataset.table`.`field`"
-  [{:keys [identifier-type components] :as identifier}]
+  [{:keys [identifier-type components], ::keys [do-not-qualify?], :as _identifier}]
   (cond
-    (::already-qualified? (meta identifier))
+    do-not-qualify?
```

Comment:
We were previously recording whether something was _already_ qualified using metadata attached to the `Identifier`. I switched this to adding a key to the `Identifier` object itself to make it easier to see when debugging stuff and harder to accidentally toss